### PR TITLE
chore: migrate Java to Kotlin spike

### PR DESCRIPTION
**Pull request by edgarvonk**
Migrate the ZTC client package of the main codebase from Java to Kotlin. 

Solves PZ-1840

### CODE REVIEW

File: src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeListParameters.kt [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import jakarta.ws.rs.QueryParam
+import java.net.URI
+
+class ZaaktypeListParameters(
+    private val catalogus: URI
+) : AbstractZTCListParameters() {
+    /**
+     * URL-referentie naar de CATALOGUS waartoe dit ZAAKTYPE behoort.
+     */
+    @QueryParam("catalogus")
```
Comment (bas-info-nl) [inline src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeListParameters.kt:16]: I think you can use (similar as done on other fields) the annotation `@field:QueryParam("catalogus")` directly in the constructor, like so:
```
class ZaaktypeListParameters(
    @field:QueryParam("catalogus") val catalogus: URI
)
```

File: src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeInformatieobjecttypeListParameters.kt [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ * SPDX-FileCopyrightText: 2022 Atos, 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.client.zgw.ztc.model
+
+import jakarta.ws.rs.QueryParam
+import java.net.URI
+
+class ZaaktypeInformatieobjecttypeListParameters(
+    private val zaaktype: URI
```
Comment (bas-info-nl) [inline src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeInformatieobjecttypeListParameters.kt:11]: I think you can use (similar as done on other fields) the annotation `@field:QueryParam("zaaktype")` directly in the constructor

File: src/main/kotlin/net/atos/client/zgw/ztc/model/StatustypeListParameters.kt [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+ * SPDX-FileCopyrightText: 2021 Atos, 2024 Lifely
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.client.zgw.ztc.model
+
+import jakarta.ws.rs.QueryParam
+import java.net.URI
+
+class StatustypeListParameters(
+    private val zaaktype: URI
```
Comment (bas-info-nl) [inline src/main/kotlin/net/atos/client/zgw/ztc/model/StatustypeListParameters.kt:11]: I think you can use (similar as done on other fields) the annotation `@field:QueryParam("zaaktype")` directly in the constructor

File: src/main/kotlin/net/atos/client/zgw/ztc/model/RoltypeListParameters.kt [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+ * SPDX-License-Identifier: EUPL-1.2+
+ */
+package net.atos.client.zgw.ztc.model
+
+import jakarta.ws.rs.QueryParam
+import net.atos.client.zgw.ztc.model.generated.RolType.OmschrijvingGeneriekEnum
+import java.net.URI
+
+class RoltypeListParameters(
+    private val zaaktype: URI,
```
Comment (bas-info-nl) [inline src/main/kotlin/net/atos/client/zgw/ztc/model/RoltypeListParameters.kt:12]: I think you can use (similar as done on other fields) the annotation `@field:QueryParam("zaaktype")` directly in the constructor.

File: src/main/java/net/atos/zac/app/zaken/converter/RESTZaaktypeConverter.java [hunk 5]
```diff
@@ -5,8 +5,8 @@
 
 package net.atos.zac.app.zaken.converter;
 
-import static net.atos.client.zgw.ztc.util.ZaakTypeUtil.isNuGeldig;
-import static net.atos.client.zgw.ztc.util.ZaakTypeUtil.isServicenormBeschikbaar;
+import static net.atos.client.zgw.ztc.util.ZaakTypeUtilKt.isNuGeldig;
```
Comment (bas-info-nl) [inline src/main/java/net/atos/zac/app/zaken/converter/RESTZaaktypeConverter.java:8]: Not sure I like that it has been renamed, but ok...

**Review by bas-info-nl — APPROVED** [review]
Up to you what to do with the comments, they're not major nor blocking as far as I'm concerned.

File: src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeListParameters.kt (hunk 1 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeListParameters.kt:16]: No, afraid not. This is exactly what was generated (in some places) and then I found out this was causing the issue that the query params were not added to the requests.. I'm not completely sure why still but it really did not work.

File: src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeInformatieobjecttypeListParameters.kt (hunk 2 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/client/zgw/ztc/model/ZaaktypeInformatieobjecttypeListParameters.kt:11]: No, see above. That did not work.. Something to look at later maybe but when adding it to the constructor the query params were not added to the requests by JAX-RS..

File: src/main/java/net/atos/zac/app/zaken/converter/RESTZaaktypeConverter.java (hunk 5 above)
Comment (edgarvonk) [reply src/main/java/net/atos/zac/app/zaken/converter/RESTZaaktypeConverter.java:8]: I don't think I can avoid that. Well, I could if I make ZaakTypeUtil a class or object but I think that is not idiomatic Kotlin. This file just contains static functions so I see no need for a class or object right?

File: src/main/kotlin/net/atos/client/zgw/ztc/model/StatustypeListParameters.kt (hunk 3 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/client/zgw/ztc/model/StatustypeListParameters.kt:11]: No, see above. That did not work..

File: src/main/kotlin/net/atos/client/zgw/ztc/model/RoltypeListParameters.kt (hunk 4 above)
Comment (edgarvonk) [reply src/main/kotlin/net/atos/client/zgw/ztc/model/RoltypeListParameters.kt:12]: No, see above. That did not work..

**Pull request merged**